src/db/gc/mod.rs

```diff
@@ -0,0 +1,217 @@
+// Copyright 2019-2023 ChainSafe Systems
+// SPDX-License-Identifier: Apache-2.0, MIT
+
+//!
+//! The current implementation of the garbage collector is concurrent mark-and-sweep.
+//!
+//! ## Design goals
+//! A correct GC algorithm that is simple and efficient for forest scenarios.
+//!
+//! ## GC Algorithm
+//! The `mark-and-sweep` algorithm was chosen due to it's simplicity, efficiency and low memory
+//! footprint. We do have to generate keys from values for some of the iterated columns due to a
```

Comment:
How ParityDB is iterated has nothing to do with the algorithm. It was mentioned in the previous GC because it directly affected the choice of algorithm.